### DISCUSSION TITLE
Why does the ImageButton width not adjust to the new width?

### DISCUSSION THREAD

**Discussion by IrisClasson** [dotnet/maui#16878] [Upgrading from Xamarin]
I'm migrating an app from Xamarin to MAUI and I'm having layout issues. One of the problems is that the ImageButton doesn't seem to use it's new width when a HeightRequest is set (and AspectFit). The image varies in size and ratio, and therefore a set width isn't an option (the code below is greatly simplified, IRL its a custom control). The original image is large and its source (ImageSource) points to images that have different sizes/aspect ratios).

The questions are:
Why is the behavior different?
How do I fix this?

Example code (original code is complex, its a custom control and we use handlers etc.)

```
<StackLayout HeightRequest="300" Padding="0,100">
    <Grid BackgroundColor="Aquamarine" HeightRequest="200">
        <Grid.ColumnDefinitions>
            <ColumnDefinition Width="auto"/>
            <ColumnDefinition Width="100"/>
        </Grid.ColumnDefinitions>
        <ImageButton Source="cup.png"
                     BackgroundColor="HotPink"
                     x:Name="btn"
                     HeightRequest="80" 
                     Aspect="AspectFit" 
                     HorizontalOptions="Start" 
                     VerticalOptions="Start">
        </ImageButton>
        <Label BackgroundColor="Coral" Text="My nice cup" Grid.Column="1"></Label>
    </Grid>
</StackLayout>
```

MAUI

![Screenshot 2023-08-18 at 14 11 42](https://github.com/dotnet/maui/assets/1671084/3d2d5f46-a811-4fd0-ae0c-510e728c300a)

Xamarin
![Screenshot 2023-08-18 at 14 10 13](https://github.com/dotnet/maui/assets/1671084/252f5fe7-8b4d-40b3-a2b6-be3b8104d427)

**Comment by xtuzy** [comment]
net8pre7 will show image
![image](https://github.com/dotnet/maui/assets/17793881/88c9778e-13e3-4b19-967a-a30052d363c0)

**Reply by IrisClasson** [reply]
Thank you! I found a workaround for 7 but will remove the workaround once we upgrade. Happy to see that this has been fixed!

**Reply by IrisClasson** [reply]
I just tested with the preview, and unfortunately the problem is still there. What seems a bit odd is that the behavior is different if I use a SVG versus a PNG, but neither will set the width to match the height. I'll dig a bit further and hopefully return with more udnerstanding.

**Reply by skha83** [reply]
I reported a similar issue last week here #16921 . It seems to be a iOS specific problem as Android works just like Xamarin did in the past. Only workaround I found was setting both width and height of the image, but that is not an option if you switch between images with different aspect ratios. This bug ruined a lot of our layouts when we converted from Xamarin to MAUI.
